packages/core/src/sessionManagement.ts

```diff
@@ -89,25 +123,46 @@ function retrieveSession(sessionCookie: CookieCache): SessionState {
   return session
 }
 
-export function persistSession(session: SessionState, cookie: CookieCache) {
+export function persistSession(session: SessionState, cookie: CookieCache, withNewSessionStrategy = false) {
+  if (withNewSessionStrategy && !utils.isEmptyObject(session)) {
+    session.expire = String(Date.now() + SESSION_EXPIRATION_DELAY)
+  }
   const cookieString = utils
     .objectEntries(session)
     .map(([key, value]) => `${key}=${value}`)
     .join(SESSION_ENTRY_SEPARATOR)
-  cookie.set(cookieString, EXPIRATION_DELAY)
+  cookie.set(cookieString, SESSION_EXPIRATION_DELAY)
 }
 
 export function stopSessionManagement() {
-  registeredActivityListeners.forEach((e) => e())
-  registeredActivityListeners = []
+  registeredListeners.forEach((e) => e())
+  registeredIntervals.forEach((interval) => clearInterval(interval))
+  registeredListeners = []
+  registeredIntervals = []
 }
 
-let registeredActivityListeners: Array<() => void> = []
+let registeredListeners: Array<() => void> = []
+let registeredIntervals: number[] = []
 
 export function trackActivity(expandOrRenewSession: () => void) {
+  const doExpandOrRenewSession = monitor(expandOrRenewSession)
   const options = { capture: true, passive: true }
   ;['click', 'touchstart', 'keydown', 'scroll'].forEach((event: string) => {
-    document.addEventListener(event, expandOrRenewSession, options)
-    registeredActivityListeners.push(() => document.removeEventListener(event, expandOrRenewSession, options))
+    document.addEventListener(event, doExpandOrRenewSession, options)
+    registeredListeners.push(() => document.removeEventListener(event, doExpandOrRenewSession, options))
+  })
+}
+
+function trackVisibility(expandSession: () => void, visibilityStateProvider: () => VisibilityState) {
```

Comment:
Could we have a `isVisible()` kind of function instead ? If we somehow need to  adapt the `Document.visibilityState` API for some specific browser, maybe it will be easier to have this logic in a separate function. Also I find it easier to read.

Reply:
> if we somehow need to adapt the Document.visibilityState API for some specific browser, maybe it will be easier to have this logic in a separate function.

Let's worry about that if we encounter this kind of issue. We would then have more insight on how to solve it.

> I find it easier to read

Update the function to use a isVisible method instead would also mean to move the `'visible'`check outside of this function which seems really appropriate to me here.

Do you have another suggestion that could keep this check and make that more legible?

Reply:
all good then